### PR TITLE
Fix typo&improve wording

### PR DESCRIPTION
**Pull request by ice1000**

### CODE REVIEW

**Comment by ice1000** [comment]
I didn't recompile it

**Comment by thautwarm** [comment]
maybe more typos could be found after you read this. PING me when it gets done.

**Comment by thautwarm** [comment]
@ice1000 As you use Rust very often, I do suggest you to read this article as soon as possible and then make your own HIGHER implementation :)

**Comment by ice1000** [comment]
> @ice1000 As you use Rust very often, I do suggest you to read this article as soon as possible and then make your own HIGHER implementation :)

No, not needing hkt atm (and because the stdlib is already not abstracted over hkts).

**Comment by ice1000** [comment]
Keyi Kanyixia dierge commit ma

File: _sources/PL/paper-reading-LHKP.rst.txt [hunk 1]
```diff
@@ -55,9 +56,9 @@ Then declare a *type class* :
 
     type 'a mappable_impl = (module Mappable with type t = 'a)
 
-Interesting, we've just implemented a type class tersely just like what is allowed in Haskell.
+Interesting, we've just implemented a type class tersely just like in Haskell.
```
Comment (thautwarm) [inline _sources/PL/paper-reading-LHKP.rst.txt:59]: 你这是我的意思吗？？

File: _sources/PL/paper-reading-LHKP.rst.txt [hunk 2]
```diff
@@ -140,17 +141,18 @@ Take care that, in last section, the so-called **LHKP** is also kind of disgusti
      map (module MapArray) ...
 
 
-It's not difficult to observe that, the type constructor like :code:`MapList` and :`MapArray` has been
-already given in other arguments:
+It's obvious that, type constructors like :code:`MapList` and :code:`MapArray` had already been
```
Comment (thautwarm) [inline _sources/PL/paper-reading-LHKP.rst.txt:144]: obvious这个，你用你的表达把我的换了，虽然没什么差别。

File: _sources/PL/paper-reading-LHKP.rst.txt [hunk 3]
```diff
@@ -140,17 +141,18 @@ Take care that, in last section, the so-called **LHKP** is also kind of disgusti
...
     val map: type t. (t mappable_impl) ('a -> 'b) -> ('a, t) app -> ('b, t) app
 
-In fact, the argument typed :code:`t mappable_impl` is unique for a given :code:`t`.
-So how about create an instance typed :code:`t mappable_impl` automatically from the type :code:`t` ?
+In fact, the argument typed :code:`t mappable_impl` is unique given a unique :code:`t`
+(in other words, :code:`mappable_impl` is injective).
+So how about create an instance of type :code:`t mappable_impl` automatically from the type :code:`t` ?
 
-If we can do this, we then have a better polymorphism which is no worse than that from Haskell.
+If we can do this, we then have a better polymorphism no worse than Haskell's.
```
Comment (thautwarm) [inline _sources/PL/paper-reading-LHKP.rst.txt:155]: 。。。

File: _sources/PL/paper-reading-LHKP.rst.txt [hunk 4]
```diff
@@ -250,8 +252,8 @@ However, using such :code:`Either` could be quite annoying:
   let () = print_string (show_either_int_int either_mapped2 ^ "\n")
 
 That sucks so much for the lack of modular implicits and, it's an internal property that
-expressing much higher kinded types(whose kind ascription is more complex than :code:`* -> *`)
-requires many other verbose codes like :code:`('c, ('b, 'a) app) app`.
+expressing much higher kinded types (whose kind ascription is more complex than :code:`* -> *`)
+requires many other boilerplate codes like :code:`('c, ('b, 'a) app) app`.
```
Comment (thautwarm) [inline _sources/PL/paper-reading-LHKP.rst.txt:256]: 类似前面有一些。boilerplate可能是一个更好的词，但是不是typo。
boilerplate会让我想到设计模式和多态。

File: _sources/PL/paper-reading-LHKP.rst.txt [hunk 5]
```diff
@@ -288,25 +290,26 @@ after introducing higher kined types.
 
     'a cons ~ 'e 't
 
-Now we don't known if :code:`cons` is a type constructor or simply an alias,
-so we cannot imply that :code:`'a ~ 'e` and :code:`cons ~ 't`, for if :code:`'cons`
+Now we don't know if :code:`cons` is a type constructor or simply an alias,
+so we cannot say :code:`'a ~ 'e` and :code:`cons ~ 't`, in case :code:`'cons`
```
Comment (thautwarm) [inline _sources/PL/paper-reading-LHKP.rst.txt:294]: 嗯，是in case。。。

Comment (thautwarm) [reply _sources/PL/paper-reading-LHKP.rst.txt:294]: in case的话，后面还要删一点东西

**Review by thautwarm — APPROVED** [review]
虽然你是我的**，但你不应该在typo的PR混进refine wording的内容

File: _sources/PL/paper-reading-LHKP.rst.txt (hunk 1 above)
Comment (ice1000) [reply _sources/PL/paper-reading-LHKP.rst.txt:59]: 不是吗

File: _sources/PL/paper-reading-LHKP.rst.txt (hunk 5 above)
Comment (ice1000) [reply _sources/PL/paper-reading-LHKP.rst.txt:294]: 是吗

File: _sources/PL/paper-reading-LHKP.rst.txt (hunk 2 above)
Comment (ice1000) [reply _sources/PL/paper-reading-LHKP.rst.txt:144]: It's easy to observe that

File: _sources/PL/paper-reading-LHKP.rst.txt (hunk 3 above)
Comment (ice1000) [reply _sources/PL/paper-reading-LHKP.rst.txt:155]: ？？？

File: _sources/PL/paper-reading-LHKP.rst.txt (hunk 4 above)
Comment (ice1000) [reply _sources/PL/paper-reading-LHKP.rst.txt:256]: noise?

File: _sources/PL/paper-reading-LHKP.rst.txt (hunk 5 above)
Comment (thautwarm) [reply _sources/PL/paper-reading-LHKP.rst.txt:294]: 是的。我这里肯定是写得for if xxx, 会怎么怎么样。in case直接说xxx就好，不用说会怎么样。

Comment (ice1000) [reply _sources/PL/paper-reading-LHKP.rst.txt:294]: 是的是的 红红太懂了

**Pull request merged**